test/interface_basic_types.test.ts

```diff
@@ -140,23 +140,86 @@ test('Test for IDL \'unsigned long long\' type', async () => {
   expect(test_interface.unsignedLongLongMethod(-1) != -1).toBe(true);
 });
 
+test('Test for IDL \'float\' type', async () => {
+  let test_interface = new bacardi.TestInterface();
+
+  // The float type is a floating point numeric type that corresponds to the set
+  // of finite single-precision 32 bit IEEE 754 floating point numbers.
+
+  expect(test_interface.floatMethod(0.0)).toBe(0.0);
+
+  const base = 1 / 2;
+  const precision = 23;
+  for (var test_case = 0; test_case < 100; test_case++) {
+    // for fraction part
+    // create a random number that has 23 bits precision
+    var fraction = 1;  // set the biggest bit.
+    for (var i = 1; i < precision; i++) {
+      if (Math.random() > 0.5)
+        fraction += base ** i;
+    }
+    fraction += base ** precision  // set the smallest bit
+
+    // for exponent part
+    // test all value of exponent part
+    // because exponent value -127 is used to express 0
+    // so this must start from -126.
+    for (var i = -126; i <= 127; i++) {
+      var float_value = fraction * (2 ** i);
+      expect(test_interface.floatMethod(float_value)).toBe(float_value);
+    }
+
+    // these two cases are beyond the range of floating-point
+    var float_value = fraction * (2 ** -127);
+    expect(test_interface.floatMethod(float_value) != float_value).toBe(true);
+    float_value = fraction * (2 ** 128);
+    expect(test_interface.floatMethod(float_value) != float_value).toBe(true);
+  }
+
+  var float_min = 2 ** (-149);
+  ;
+  expect(test_interface.floatMethod(float_min)).toBe(float_min);
+  // the value beyond the range of exponent
+  expect(test_interface.floatMethod(float_min / 2) != float_min / 2).toBe(true);
+  expect(test_interface.floatMethod(float_min / 2) != float_min / 2).toBe(true);
+});
+
 test('Test for IDL \'double\' type', async () => {
   let test_interface = new bacardi.TestInterface();
 
   // The double type is a floating point numeric type that corresponds to the
   // set of finite double-precision 64 bit IEEE 754 floating point numbers.
+
   expect(test_interface.doubleMethod(0.123456789012345))
       .toBe(0.123456789012345);
 
-  // FIXME(zino): We should test for comparing single-precision floating point.
-
-  // FIXME(zino): We should check whether the type is restricted or
-  // unrestricted.
+  const base = 1 / 2;
+  const precision = 52;
+  for (var test_case = 0; test_case < 100; test_case++) {
```

Comment:
Please use `let` instead of `var`. The `var` is function scope and `let` is block scope.
So, we prefer to use `let` to avoid unexpected behavior.